ludwig/api.py

```diff
@@ -607,6 +607,10 @@ def on_epoch_end(self, trainer, progress_tracker, save_path):
                     if not path_exists(weights_save_path):
                         with open_file(weights_save_path, "wb") as f:
                             torch.save(self.model.state_dict(), f)
+                    # Adds a flag to all input features indicating that the weights are saved in the checkpoint.
+                    for input_feature in self.config["input_features"]:
+                        input_feature["saved_weights_in_checkpoint"] = True
```

Comment:
Would generalize it to all output features and combiner as well

Reply:
Maybe we could have a separate pr for it though, we can add to the board as a reminder

Reply:
I would doublecheck the placement of this code for setting it to true, need to make sure it is hit by every possible path that involves training (cli, api, train, experiment, etc.)

Reply:
I thought about that, and I was curious how you would implement this: either trainer needs to inform LudwigModel when it saves,

Reply:
Oh hey what if we move saving down to `LudwigModule`.  Define load and save at that level, implement it for ECD only for today.  That is a push in the direction of modular saving and loading.  On the downside, its a bigger refactor than I wanted to do here to address this one bug.

Reply:
Moved the setting of the flag to the first save_config before training starts (instead of after training).

I would like to push saving logic down to the model (ECD), will do that in a separate PR.  Opened an issue for this: https://github.com/ludwig-ai/ludwig/issues/1909